src/instrumentor.cpp

```diff
@@ -1056,6 +1029,19 @@ void instrumentor::instrument()
             exit(1);
         }
 
+        // If using C++ API, check that config file contains code for scoped instrumentation
+        if (use_cxx_api) {
+            if (ryml::ConstNodeRef mainInsertScope = yaml_tree["main_insert_scope"]; mainInsertScope.invalid()) {
+               llvm::errs() << "Using C++ Instrumentation API requires `main_insert_scope` in config file.\n";
+                exit(2);
+            }
+            if (ryml::ConstNodeRef functionBeginScope = yaml_tree["function_begin_insert_scope"]; functionBeginScope.
+                invalid()) {
+                llvm::errs() << "Using C++ Instrumentation API requires `function_begin_insert_scope` in config file.\n";
+                exit(2);
+            }
+        }
```

Comment:
Good idea to put some checks in. We should probably add (or verify that it's present already) checks for the other required config file components at some point.